### PR TITLE
chore: upgrade to spring-boot 2.7.8, with springdoc instead of springfox

### PR DESCRIPTION
**Pull request by guilgaly**
Springfox is no longer maintained, and does not support spring-boot 2.6+. The recommended upgrade path is to switch to Springdoc.

### CODE REVIEW

File: src/main/java/io/gatling/demostore/WebConfig.java [hunk 1]
```diff
@@ -14,11 +14,6 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
                 .addResourceHandler("/resources/**")
                 .addResourceLocations("classpath:/media/")
                 .addResourceLocations("classpath:/downloads/");
-
-        registry
-                .addResourceHandler("/swagger-ui/**")
-                .addResourceLocations("classpath:/META-INF/resources/webjars/springfox-swagger-ui/")
-                .resourceChain(false);
```
Comment (guilgaly) [inline src/main/java/io/gatling/demostore/WebConfig.java:21]: Doesn't seem needed, I assume `springdoc-openapi-ui` handles this automagically 🤷

File: src/main/java/io/gatling/demostore/api/SwaggerConfig.java [hunk 2]
```diff
@@ -1,67 +1,35 @@
...
-                Collections.singletonList(new SecurityReference("JWT", authorizationScopes));
-        return SecurityContext.builder()
-                .securityReferences(securityReferences)
-                .operationSelector(this::isSecuredOperation)
-                .build();
-    }
-
-    private boolean isSecuredOperation(OperationContext context) {
-        return context.httpMethod() != HttpMethod.GET
-                && !context.getName().equals("createAuthenticationToken");
```
Comment (guilgaly) [inline src/main/java/io/gatling/demostore/api/SwaggerConfig.java:65]: No idea if it's still possible to define a global filter like this, but probably makes more sense anyway to use `@Operation(security = ...)` on the relevant routes.

File: src/main/java/io/gatling/demostore/WebConfig.java (hunk 1 above)
Comment (slandelle) [reply src/main/java/io/gatling/demostore/WebConfig.java:21]: 🤷

File: src/main/java/io/gatling/demostore/api/SwaggerConfig.java (hunk 2 above)
Comment (slandelle) [reply src/main/java/io/gatling/demostore/api/SwaggerConfig.java:65]: No, it's no longer possible. Local `@Operations` are the way to go. Then I didn't check how override was working. Does an annotation completely overrides? Or just the defined attributes?

**Review by slandelle — APPROVED** [review]

**Review by notdryft — APPROVED** [review]

**Pull request merged**